server/api.go

```diff
@@ -690,3 +700,101 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request) {
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
+
+	ctx := context.Background()
+
+	var githubClient *github.Client
+
+	if info, err := p.getGitHubUserInfo(userID); err != nil {
+		writeAPIError(w, err)
+		return
+	} else {
```

Comment:
1/5 remove the `else` after `return` - idiomatic go.

Reply:
@dhadiseputro, I also suggest removing the else statement, but also the predeclaration of `githubClient`  There are a couple of additional changes in the suggestion below.  

* changed the inline error check to multiline.
* renamed `err` to `apiErr` for returned error from `p.getGitHubUserInfo()` call

```go 
info, apiErr := p.getGitHubUserInfo(userID)
if apiErr != nil {
	writeAPIError(w, apiErr)
	return
}
githubClient := p.githubConnect(*info.Token)
```